### PR TITLE
add tidbinitializer crd

### PR DESCRIPTION
**Pull request by aylei**
Signed-off-by: Aylei <rayingecho@gmail.com>

#1262 
### Does this PR introduce a user-facing change?:
 
 ```release-note
Add new CRD TidbInitializer.
 ```
@Yisaer @tennix @sdojjy @DanielZhangQD @LinuxGit PTAL

### CODE REVIEW

File: pkg/apis/pingcap/v1alpha1/tidbinitializer_types.go [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+	// Spec defines the desired state of TidbInitializer
+	Spec TidbInitializerSpec `json:"spec"`
+
+	// +k8s:openapi-gen=false
+	// Most recently observed status of the TidbInitializer
+	Status TidbInitializerStatus `json:"status"`
+}
+
+// +k8s:openapi-gen=true
+// TidbInitializer spec encode the desired state of tidb initializer Job
```
Comment (DanielZhangQD) [inline pkg/apis/pingcap/v1alpha1/tidbinitializer_types.go]: Also need o add `permitHost`, `initSqlConfigMapName`.

Comment (Yisaer) [reply pkg/apis/pingcap/v1alpha1/tidbinitializer_types.go]: +1,  `initSqlConfigMapRef` is used widely in `initSql` case.

Comment (aylei) [reply pkg/apis/pingcap/v1alpha1/tidbinitializer_types.go]: Is it still necessary if we change the type of `initSql` to string?

Comment (aylei) [reply pkg/apis/pingcap/v1alpha1/tidbinitializer_types.go]: I will add it to make the functionality consistent with helm

**Comment by aylei** [comment]
@DanielZhangQD @Yisaer PTAL

File: pkg/apis/pingcap/v1alpha1/tidbinitializer_types.go [hunk 2]
```diff
@@ -53,21 +54,27 @@ type TidbInitializer struct {
 // +k8s:openapi-gen=true
 // TidbInitializer spec encode the desired state of tidb initializer Job
 type TidbInitializerSpec struct {
+	Image string `json:"image"`
```
Comment (DanielZhangQD) [inline pkg/apis/pingcap/v1alpha1/tidbinitializer_types.go]: The `Image` and `ImagePullPolicy` are to cover below configs in `values.yaml`, right?
```
# mysqlClient is used to set password for TiDB
# it must has Python MySQL client installed
mysqlClient:
  image: tnir/mysqlclient
  imagePullPolicy: IfNotPresent
```
We also have to cover the initializer.resources config:
```
  initializer:
    resources: {}
    # limits:
    #  cpu: 100m
    #  memory: 100Mi
    # requests:
    #  cpu: 100m
    #  memory: 100Mi
```

Comment (aylei) [reply pkg/apis/pingcap/v1alpha1/tidbinitializer_types.go]: Good catch!

File: charts/tidb-operator/templates/controller-manager-rbac.yaml [hunk 3]
```diff
@@ -49,15 +49,7 @@ rules:
...
-  resources:
-  - tidbclusters
-  - tidbclusters/finalizers
-  - backups
-  - backups/finalizers
-  - backupschedules
-  - backupschedules/finalizers
-  - restores
-  - restores/finalizers
+  resources: ["*"]
```
Comment (aylei) [inline charts/tidb-operator/templates/controller-manager-rbac.yaml]: Grant all resources of our own group should be safe

**Pull request merged**